src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -597,6 +606,91 @@ private ProcessWaitState GetWaitState()
             return _waitStateHolder._state;
         }
 
+        private static (uint userId, uint groupId) GetUserAndGroupIds(ProcessStartInfo startInfo)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(startInfo.UserName));
+
+            (uint? userId, uint? groupId) = GetUserAndGroupIds(startInfo.UserName);
+
+            if (userId == null || groupId == null)
+            {
+                throw new Win32Exception(SR.Format(SR.UserDoesNotExist, startInfo.UserName));
```

Comment:
Are there situations where userId might be non-null but groupId null?  The error message here says that the user doesn't exist, so I'm wondering if userId will be null iff groupId is null, in which case the check could be simplified to just check userId, maybe with an assert on groupId.

Reply:
+1. Also check in line 296?

Reply:
No, they are either both set, or both null.  I'll check just `userId` and add an Assert for `groupId`.